### DISCUSSION TITLE
ImageGlass doesn't auto-update the viewing image when modified / rotated

### DISCUSSION THREAD

**Discussion by MxMalk** [d2phap/ImageGlass#337] [General] [answered]
It would be nice to recall frmMain.Prepare() after saving changes to an image and on refreshing and maybe even have an radio toggle option in the menu to "Re-Prepare" after a chosen amount of minutes/seconds.

One issue would be the load time on big files and folders. Maybe measure the runtime of the first call?

I also see an issue with the use-case in that one wants to rotate (and save) a lot of pictures quickly. Calling Prepare on every rotation could slow this down a lot. Maybe only reload after a least 2000ms of user inactivity and if runtime is below 1000ms?

**One solution could be:** 
Having a "Live-Mode" and a "Lazy-Mode":
 - Live-Mode reloading on Save, Refresh, and also after a certain time. 
 - Lazy-Mode being the default and reloading only on file opening and maybe on Refresh.

**Comment by d2phap** [comment, marked as answer]
Hi @MxMalk 
What version did you try?
Can you check the latest build http://imageglass.org/moon?
It will auto reload the image if it is changed (rotation / modified...)

**Comment by MxMalk** [comment]
Sorry, thats embarrassing, yes i really had an older source on my system. 

Thank you for your great work. ImageGlass is an awesome tool.

**Comment by d2phap** [comment]
No problem @MxMalk 
It's not officially released yet 😄 but will do soon. Please check for update this Monday
